table/table.cc

```diff
@@ -111,46 +109,49 @@ void Table::ReadMeta(const Footer& footer) {
 
   Iterator* iter = meta->NewIterator(BytewiseComparator());
 
-  bool found,first;
-  const FilterPolicy * policy, * next;
-  std::string key;
-
-  first=true;
-  next=NULL;
+  // read filter only if policy set
+  if (NULL != rep_->options.filter_policy) {
+      bool found,first;
+      const FilterPolicy * policy, * next;
 
-  do
-  {
-      found=false;
+      first=true;
+      next=NULL;
 
-      if (first)
-      {
-          policy=rep_->options.filter_policy;
-          next=FilterInventory::ListHead;
-          first=false;
-      }   // if
-      else
+      do
       {
-          policy=next;
-          if (NULL!=policy)
-              next=policy->GetNext();
-          else
-              next=NULL;
-      }   // else
+          found=false;
 
-      if (NULL!=policy)
-      {
-          key = "filter.";
-          key.append(policy->Name());
-          iter->Seek(key);
-          if (iter->Valid() && iter->key() == Slice(key))
+          if (first)
           {
-              ReadFilter(iter->value(), policy);
-              gPerfCounters->Inc(ePerfBlockFilterRead);
-              found=true;
+              policy=rep_->options.filter_policy;
+              next=FilterInventory::ListHead;
+              first=false;
           }   // if
-      }   //if
-  } while(!found && NULL!=policy);
+          else
+          {
+              policy=next;
+              if (NULL!=policy)
+                  next=policy->GetNext();
+              else
+                  next=NULL;
+          }   // else
 
+          if (NULL!=policy)
+          {
+              key = "filter.";
+              key.append(policy->Name());
+              iter->Seek(key);
+              if (iter->Valid() && iter->key() == Slice(key))
+              {
+                  ReadFilter(iter->value(), policy);
+                  gPerfCounters->Inc(ePerfBlockFilterRead);
+                  found=true;
+              }   // if
+          }   //if
+      } while(!found && NULL!=policy);
+  }   // if
+
+  // always read counters
   key="stats.sst1";
```

Comment:
Is this a key in the user-visible key space? If so, I worry that it is not unique enough to avoid collision with a user key.

The same concern applies to the "filter." key above.